unstructured/partition/pptx.py

```diff
@@ -58,97 +67,235 @@ def partition_pptx(
     include_page_breaks
         If True, includes a PageBreak element between slides
     metadata_filename
-        The filename to use for the metadata. Relevant because partition_ppt converts the
-        document .pptx before partition. We want the original source filename in the
-        metadata.
+        The filename to use for the metadata. Relevant because partition_ppt() converts its
+        (legacy) .ppt document to .pptx before partition. We want the filename of the original
+        .ppt source file in the metadata.
     metadata_last_modified
         The last modified date for the document.
-
-
     include_slide_notes
         If True, includes the slide notes as element
     """
-
-    # Verify that only one of the arguments was provided
+    # -- verify only one source-file argument was provided --
     exactly_one(filename=filename, file=file)
-    last_modification_date = None
-    if filename is not None:
-        if not filename.startswith("/tmp"):
-            last_modification_date = get_last_modified_date(filename)
-
-        presentation = pptx.Presentation(filename)
-    elif file is not None:
-        last_modification_date = get_last_modified_date_from_file(file)
-        presentation = pptx.Presentation(
-            spooled_to_bytes_io_if_needed(
-                cast(Union[BinaryIO, SpooledTemporaryFile], file),
-            ),
+
+    if isinstance(file, SpooledTemporaryFile):
+        file.seek(0)
+        file = io.BytesIO(file.read())
```

Comment:
If the spooled temporary file is being loaded into memory here, doesn't that defeat the point of using spooled temporary files? Also, if spooled temporary file typing has been removed from the file kwargs, why would we need to check the instance of file? are we expecting users to inject a spooled temporary file anyway?

Reply:
I sure thought so! :)

But in speaking with Crag, apparently `SpooledTemporaryFile` has exploded for us in the past when the file is huge (GBs size) and the Python version is below some threshold. So apparently the conversion is legitimately required.

This is essentially what `spooled_to_bytes_io_if_needed()` does, but that function needs some work and is used in a lot of other places so I thought for the cost of a couple lines I'd just pull out the needful here with the idea we might extract a "file-loader" object at some point that would encapsulate this and many of the surrounding lines here.

Reply:
Makes sense, could you add an inline todo/note that summarizes that?

Reply:
k, turns out this was put in for a particular downstream package (maybe PDF-miner or something) and just got rubber-stamped around it looks like. I'm cutting it out as `python-docx` and `python-pptx` like `SpooledTemporaryFile` just fine and do exactly what the `spooled_to_bytes_io_if_needed()` function does which is read the whole thing once from the top. I am going to leave in the `seek(0)` though (when `file` has `.seek()`), just in case.

Reply:
Hmm, okay, I think I discovered the reason and it's that `python-docx` and `python-pptx` both use ZipFile (because the format is based on a zip-archive), and `ZipFile` checks for a `seekable` attribute that `SpooledTemporaryFile` doesn't have. So it turns out to be a real problem after all. And it makes the comment I need to add clear, so that's a plus :)